many general changes & enhancements

Carandiru0 opened the pull request:
many changes, some specialized for application specific (minCity development), many general changes / enhancements
not all changes can be merged, so if you "pick-and-choose" what to keep and what to ignore of the changes I've made, there's some nice stuff in here. It's all in the fork I've made under "initial commit"

vku.hpp vkuframework.hpp vkuaddon.hpp vkudoublebuffer.h

Thanks for Vookoo!

Cheers!!

Carandiru0 committed:
initial commit

many changes, some specialized for application specific (minCity development), many general changes / enhancements

pocdn commented:
I have successfully used vookoo on both my linux and windows (gcc on mingw64) prior to this proposed branch. This branch appears to inject non-standard cross-platform dependencies to the point where I can't get it to compile. In particular, I'm getting errors "no such file" on the includes for <Utility/mem.h>, <fmt/fmt.h>, <Utility/enum.h> if not others. What compiler and OS was this branch tested on? Hopefully we can find away to keep vookoo cross-platform and compiler/OS agnostic.

Carandiru0 commented:
@pocdn Take what you actually want to use out of it and not use what you don't want to use. No automatic merge here, these changes can be compatible in some cases but in others not (like integration and usage of the vma allocator lib). There are also a lot of specialization to support what I'm using it for. But I'd rather make a contribution where you can pick and choose what you'd like to keep, improvements, etc.   vs not making any contribution at all right? So maybe the implementation of Specialization Constants could be used, or the extended image classes, etc. Multisampling?

lhog commented:
Thanks for sharing your work. As @pocdn witnessed and you suggested it's going to be hard to merge the whole scope.

I think we can leave this PR open "forever", and you can add new stuff you're developing in-house too. This might serve as a good source of code for those trying to implement things similar to what you have done.

As far as merging I can suggest breaking this PR into parts (if possible), so each such improvement can be reviewed and merged individually.

pocdn commented:
Definitely thanks for sharing the code. I understand now the idea is to use this PR to learn from and pick/choose parts from your contributed code.

Carandiru0 committed:
fixes for all ....

synchronization errors, gone.
validation errors gone.
resolve was cause for corruption in empty subpass, collapsed to renderpass.
nvidia compatability, getting there, have reflections. Need to do transparency (intersection of transparent and opaque voxels corrupted). Flickering lighting, seems to be dependent on frametime average. more bugs togo... at least it now outputs something.
